### PR TITLE
step to volmdlr

### PR DESCRIPTION
**Pull request by GabrielJMS**
* **Please check if the PR fulfills these requirements**
- [x] The commit messages follows our guidelines (explicit, in english)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [x] CHANGELOG.md has been updated

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
- [x] Bug fix:
- [ ] New features:
- [ ] Documentation
- [x] Optimization

* **Other information**:

### CODE REVIEW

File: volmdlr/faces.py [hunk 1]
```diff
@@ -2629,21 +2767,38 @@ def fullarc3d_to_2d(self, fullarc3d):
...
-            p2 = p1 + volmdlr.TWO_PI * volmdlr.Y2D
-            return [vme.LineSegment2D(p1, p2)]
-
-        points = [self.point3d_to_2d(p) for p in fullarc3d.discretization_points(angle_resolution=5)]
+        if math.isclose(self.frame.w.dot(fullarc3d.normal), 0, abs_tol=1e-4):
+            theta_plus_pi = theta1 + math.pi
+            if theta1 > theta3:
+                theta_plus_pi = theta1 - math.pi
+            elif theta1 < theta3:
+                theta_plus_pi = theta1 + math.pi
```
Comment (AxelRinghausen) [inline volmdlr/faces.py:2786]:
```suggestion
            if theta1 > theta3:
                theta_plus_pi = theta1 - math.pi
            else:
                theta_plus_pi = theta1 + math.pi
```

File: volmdlr/faces.py [hunk 2]
```diff
@@ -2629,21 +2767,38 @@ def fullarc3d_to_2d(self, fullarc3d):
...
+            half_pi = 0.5 * math.pi
+            if phi1 > phi3:
+                half_pi = 0.5 * math.pi
+            elif phi1 < phi3:
+                half_pi = -0.5 * math.pi
```
Comment (AxelRinghausen) [inline volmdlr/faces.py:2791]:
```suggestion
            if phi1 > phi3:
                half_pi = 0.5 * math.pi
            else:
                half_pi = -0.5 * math.pi
```

**Review by AxelRinghausen — APPROVED** [review]

File: volmdlr/faces.py [hunk 3]
```diff
@@ -1310,7 +1310,133 @@ def rectangular_cut(self, x1: float, x2: float,
...
+
+        length = arc3d.length()
+        theta3, _ = self.point3d_to_2d(arc3d.point_at_abscissa(0.001 * length))
+        theta4, _ = self.point3d_to_2d(arc3d.point_at_abscissa(0.98 * length))
+
+        # make sure that the references points are not undefined
+        if abs(theta3) == math.pi:
+            theta3, _ = self.point3d_to_2d(arc3d.point_at_abscissa(0.002 * length))
+        if abs(theta4) == math.pi:
+            theta4, _ = self.point3d_to_2d(arc3d.point_at_abscissa(0.97 * length))
```
[the remark this answers is not in the document]
Comment (GabrielJMS) [reply volmdlr/faces.py:1401]: it's only when the absolute value of theta is exactly pi that we don't know if it should be -pi or pi

File: volmdlr/faces.py [hunk 4]
```diff
@@ -1310,7 +1310,133 @@ def rectangular_cut(self, x1: float, x2: float,
...
+        end = self.point3d_to_2d(fullarc3d.end)
+
+        theta3, _ = self.point3d_to_2d(fullarc3d.point_at_abscissa(0.001 * length))
+        theta4, _ = self.point3d_to_2d(fullarc3d.point_at_abscissa(0.98 * length))
+
+        # make sure that the references points are not undefined
+        if abs(theta3) == math.pi:
+            theta3, _ = self.point3d_to_2d(fullarc3d.point_at_abscissa(0.002 * length))
+        if abs(theta4) == math.pi:
+            theta4, _ = self.point3d_to_2d(fullarc3d.point_at_abscissa(0.97 * length))
```
[the remark this answers is not in the document]
Comment (GabrielJMS) [reply volmdlr/faces.py:1423]: it's only when the absolute value of theta is exactly pi that we don't know if it should be -pi or pi

[the remark this answers is not in the document]
Comment (AxelRinghausen) [reply volmdlr/faces.py:1684]: But if `theta3 = math.pi - 1e-9` you are fine with `abs(theta3) != math.pi`?

**Pull request merged**